src/Shared/Model/Enums/PackageRemovalReason.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model.Enums;
+
+/// <summary>
+/// Enum for information about when a package no longer exists on a package manager.
+/// </summary>
+public enum PackageRemovalReason
+{
+    /// <summary>
+    /// When the version was unpublished from the repository/registry by someone.
+    /// </summary>
+    VersionUnpublished,
```

Comment:
Should we have different Enums when checking package existence vs. package version existence? I know there would be some duplication, but if a user asks OSSGadget about the existence of a package, that method would never return `VersionUnpublished`, correct? As this PR stands, the type system would not help the user avoid checking for an impossible value, as it could if there were separate enums for package vs. package version

Reply:
Sounds good! Addressed and resolving.